### PR TITLE
Add CannoliCommand and CannoliAlignerCommand.

### PR DESCRIPTION
**Pull request by heuermh**
Adding CannoliCommand and CannoliAlignerCommand, work in progress towards #34.  I get a little shaky with the Scala companion stuff, let me know if this looks like the right approach.

### CODE REVIEW

File: src/main/scala/org/bdgenomics/cannoli/CannoliCommand.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  @Args4jOption(required = false, name = "-use_docker", usage = "If true, uses Docker to launch the executable. If false, uses the executable path.")
+  var useDocker: Boolean = false
+
+  // must be defined due to ADAMSaveAnyArgs, but unused here
+  var sortFastqOutput: Boolean = false
+}
+
+abstract class CannoliCommand[A <: CannoliArgs] extends BDGSparkCommand[A] with Logging {
+  // todo: various docker stuff
```
Comment (fnothaft) [inline src/main/scala/org/bdgenomics/cannoli/CannoliCommand.scala]: Just flagging this as a TODO.

File: src/main/scala/org/bdgenomics/cannoli/CannoliCommand.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          "Required index file %s%s does not exist.".format(indexPath, ext))
+      })
+    })
+
+    val optionalPathsWithScheme = optionalExtensions.flatMap(optionalPath)
+
+    pathsWithScheme ++ optionalPathsWithScheme
+  }
+
+  // todo: various index stuff
```
Comment (fnothaft) [inline src/main/scala/org/bdgenomics/cannoli/CannoliCommand.scala]: Just flagging this as a TODO as well.

**Comment by heuermh** [comment]
> Pretty sure this is just a WIP ;)

Yup, looking for feedback on whether these abstract classes might be the right approach.

File: src/main/scala/org/bdgenomics/cannoli/CannoliCommand.scala [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+abstract class CannoliAlignerCommand[A <: CannoliAlignerArgs] extends CannoliCommand {
+
+  protected def getIndexPaths(
+    sc: SparkContext,
+    indexPath: String,
+    requiredExtensions: Seq[String],
+    optionalExtensions: Seq[String]): Seq[String] = {
+
+    // oddly, the hadoop fs apis don't seem to have a way to do this?
+    def canonicalizePath(fs: FileSystem, path: Path): String = {
```
[the remark this answers is not in the document]
Comment (heuermh) [reply src/main/scala/org/bdgenomics/cannoli/CannoliCommand.scala]: idk, you wrote that comment :)

**Comment by heuermh** [comment]
Closing in favor of approach in #69.

**Pull request closed without merging**